frontend/projects/upgrade/src/app/features/dashboard/experiment-users/components/stratification-factor-list/stratification-factor-list.component.ts

```diff
@@ -48,8 +52,19 @@ export class StratificationComponent implements OnInit {
             : key + '=' + stratificationFactor.factorValue[key];
           totalUsers += stratificationFactor.factorValue[key];
         });
+
         factorSummary = totalUsers.toString() + ' ' + factorSummary + ' (' + tempSummary + ')';
-        return { factor: stratificationFactor.factor, summary: factorSummary };
+        const status =
```

Comment:
I think the enums are not useful here? we just need a boolean. I think we've got some things we can smooth out here:

The intent of `stratificationFactor.experimentIds[0] !== null` looks to be an existence check for at least one id, which will fail if it's `undefined` for instance, which seems possible. `!stratificationFactor.experimentIds[0]` would be safer to catch all falsey values, or even better is probably `stratificationFactor?.experimentIds.some((experimentId) => experimentId);` to truly capture what you're really wanting: if any member of experimentIds is truthy, then the "status" is true.

I think you can eliminate these enums and change `checkStratificationFactorUsageStatus`, they are only used in the frontend and only here. Then you also don't need to call `checkStratificationFactorUsageStatus` in the html, you can just check the boolean.

Given the above, since `status` is really at the end of the day a boolean indicator, it would be more clear if we called this `isUsed` on the `StratificationFactorTableRows` type interface. `status: true` would not be obvious if you didn't already know what it meant.

In summary, here are the suggestions (I have not checked this, i'm just coding in the comment box, this is just a suggestion):

```
interface StratificationFactorsTableRow {
  factor: string;
  summary: string;
  isUsed: boolean;
  experimentIds: string[];
}
```

```
checkStratificationFactorUsageStatus(experimentIds: string[]) {
   if (Array.isArray(experimentIds) {
       return experimentIds.some(id => id);
   }
   
   return false;
}
```

return {
          factor: stratificationFactor.factor,
          summary: factorSummary,
          status: this.checkStratificationFactorUsageStatus(stratificationFactor.experimentIds);
          experimentIds: stratificationFactor.experimentIds,
        };
```

Reply:
We already use a similar enum for segment usage, so keeping it in sync. Should I go ahead and make changes as per your suggestion, as the enum has kind of 2 values only, which can definitely handled with a true/false boolean.

<img width="456" alt="Screenshot 2023-10-23 at 1 23 36 PM" src="https://github.com/CarnegieLearningWeb/UpGrade/assets/33730817/8e75daf6-833d-455e-8f80-a08410873236">

Reply:
If it's only the two possible states, which are basically on/off, I think it would make sense to do it as a boolean, would be simpler to me. What do the 'locked' and 'unlocked' states of the segments mean? I don't think I've seen that.

Reply:
Okay, will have boolean for this. The segment 'locked' and 'unlocked' states was mainly to lock the edit mode but used/unused enums do that now. So, will remove that part as well in a separate PR.

Reply:
@danoswaltCL Done. Can you please review again?